client.js

```diff
@@ -250,6 +266,74 @@ class Client {
     this.ws.send(buffer);
   }
 
+  /**
+   * Helper method to build a proper EventQuery message from a simple plain object.
+   *
+   *
+   *
+   *
+   * @param {Object} query - The simple plain object query.
+   * @returns {DBMessaging.Protobuf.EventQuery} - The built EventQuery message.
+   */
+  _buildEventQuery(query) {
+    const root = require('./generated/containerPb.js');
+    const { EventQuery } = root.DBMessaging.Protobuf;
+    const { MatchType } = EventQuery;
+
+    // Create base query with primitive fields
+    const baseQuery = {
+      timeRangeBegin: query.timeRangeBegin || 0,
+      timeRangeEnd: query.timeRangeEnd || Math.floor(Date.now() / 1000),
+      codeMask: query.codeMask !== undefined ? query.codeMask : 0xFFFFFFFF,
+      limit: query.limit || 50,
```

Comment:
Maybe just remove the field when user has not set it and let the server decide how many replies to send?

All query options in the protocol are optional I think, including timeRangeBegin and timeRangeEnd